src/claimtrie.cpp

```diff
@@ -2524,13 +2535,30 @@ bool CClaimTrieCache::clear() const
     supportExpirationQueueCache.clear();
     namesToCheckForTakeover.clear();
     cacheTakeoverHeights.clear();
+    claimsToDelete.clear();
     return true;
 }
 
 bool CClaimTrieCache::flush()
 {
     if (dirty())
         getMerkleHash();
+
+    // Delete claims first so that if an entry was deleted and then
+    // later re-added, it will first be deleted here, and then
+    // re-added in the processing after this
+    if (!claimsToDelete.empty())
+    {
+        for(claimIndexClaimListType::iterator it = claimsToDelete.begin(); it != claimsToDelete.end(); ++it)
+            base->removeFromClaimIndex(*it);
+        claimsToDelete.clear();
+    }
+    if (!claimsToAdd.empty())
```

Comment:
Do "updated" claims have the same problem?

Reply:
No.